posthog/api/github_sdk_versions.py

```diff
@@ -48,3 +80,132 @@ def github_sdk_versions(request: Request) -> JsonResponse:
             response[sdk_type] = {"error": "SDK data not available. Please try again later."}
 
     return JsonResponse(response)
+
+
+@api_view(["GET"])
+@permission_classes([IsAuthenticated])
+def sdk_version_date(request: Request, sdk_type: str, version: str) -> JsonResponse:
+    """
+    Lazy-load release dates for versions not in main cache.
+    Fetches from GitHub on-demand and caches permanently (release dates are immutable).
+    """
+    user = cast(User, request.user)
+    if not posthoganalytics.feature_enabled("sdk-doctor-beta", str(user.distinct_id)):
+        raise exceptions.ValidationError("SDK Doctor is not enabled for this user")
+
+    # Validate SDK type
+    if sdk_type not in SDK_REPO_MAP:
+        raise exceptions.ValidationError(f"Invalid SDK type: {sdk_type}")
+
+    redis_client = get_client()
+    cache_key = f"github:sdk_version_date:{sdk_type}:{version}"
+
+    # Check cache first
+    cached_date = redis_client.get(cache_key)
+    if cached_date:
```

Comment:
When writing code that might be cached you should always follow this format

```
cached_data = get_from_cache()
if not cached_date:
  cached_data = get_cached_data()

return cached_data
```



This way you don't need to return a `JsonResponse`​ twice, the return statement is always non-conditional